Source/DafnyCore/Verifier/Translator.cs

```diff
@@ -38,6 +38,15 @@ public partial class Translator {
     public bool UseOptimizationInZ3 { get; set; }
 
     void AddOtherDefinition(Bpl.Declaration declaration, Axiom axiom) {
+      sink.AddTopLevelDeclaration(axiom);
+
+      // Axioms that have triggers and bound variables do not need to be inside
+      // uses clauses. Putting such axioms inside uses clauses weakens pruning
+      // when the trigger contains more than one function or constant symbol combined.
+      if (axiom.Expr is Microsoft.Boogie.QuantifierExpr qe && qe.Dummies != null && qe.Dummies.Any() &&
+          qe.Triggers != null && qe.Triggers.Tr != null && qe.Triggers.Tr.Any()) {
+        return;
```

Comment:
Can you add a comment that gives an example of when this return would trigger for Dafny?

Reply:
I just did.

I should note that, this will not return early when the outermost expression of an axiom is not quantified. This seems to happen in practice in axioms like `k < SomeConstant ==> forall ...`, so these axioms will still be in `uses` clauses. I do not really understand what happens at solver level with such axioms, so I did not want to change this.